lib/pbench/test/unit/agent/test_tool_group.py

```diff
@@ -269,3 +270,52 @@ def mock_read_text(path):
                 assert tg.get_label(host) == labels[host]
             assert tg.get_tools("bad-host") == {}
             assert tg.get_label("bad-host") == ""
+
+
+def test_gen_tool_groups():
+    """Verify the tool group directory generator."""
+
+    class MockPath:
+        """A simple ToolGroup mock to generate host names from different
+        tool group directories.
+        """
+
+        _groups = set(("heavy", "light", "medium"))
```

Comment:
"_groups" -->  "_sets" (or perhaps "_toolsets" to avoid ambiguity)?

Reply:
They are not tool sets but names of tool groups; perhaps `_names`?